File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.py
def solution(arr):
    # 가장 작은 원소 찾기
    min_num = arr[0]
    for i in range(1,len(arr)):
        min_num = min(min_num, arr[i])
    # 제거
    arr.remove(min_num)
    # 비어있으면 [-1] 리턴
    return arr if arr else [-1]

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public int[] solution(int[] arr) {
        // arr를 순회하면서 최솟값을 찾기 만약 같다면 계속 채워넣기 
        final int min = Arrays.stream(arr)
            .min()
            .getAsInt();
        
        int[] removedArray = Arrays.stream(arr)
            .filter(num -> num != min)
            .toArray();
        for(int num: removedArray){
            log(num);   
        }
        
        return removedArray.length == 0 ? new int[]{-1} : removedArray;
    }
}